Yosemite/Yosemite/PointOfSale/Items/PointOfSaleBarcodeScanService.swift

```diff
@@ -41,13 +47,19 @@ public final class PointOfSaleBarcodeScanService: PointOfSaleBarcodeScanServiceP
     /// Looks up a POSItem using a barcode scan string
     /// - Parameter barcode: The barcode string from a scan (global unique identifier)
     /// - Returns: A POSItem if found, or throws an error
-    public func getItem(barcode: String) async throws -> POSItem {
+    public func getItem(barcode: String) async throws(PointOfSaleBarcodeScanError) -> POSItem {
```

Comment:
How did I not know we could specify what error type is thrown... is this compliler checked?

This works well for our needs here, good choice... but I think in many places it's preferable to throw the unchanged error to the app layer, so that we can log it without losing the context. `underlyingError` works fine, but might not be shown in the logs/tracks.